src/main/scala/com/gu/newsletterlistcleanse/sqs/ParseSqsMessage.scala

```diff
@@ -0,0 +1,21 @@
+package com.gu.newsletterlistcleanse.sqs
+
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import io.circe
+import io.circe.Decoder
+import io.circe.parser.decode
+import scala.collection.JavaConverters._
+import com.gu.newsletterlistcleanse.EitherConverter.EitherList
+
+
+
+object ParseSqsMessage {
+
+  def apply[T: Decoder](sqsEvent: SQSEvent): Either[List[circe.Error], List[T]] = {
```

Comment:
Tiny nitpick, as here it becomes a matter of preference and various developers will have varying opinions. So bear in mind what you have written is not only perfectly valid, but some devs would encourage you to using this pattern.

I personally prefer avoiding the `apply` method, except for construction of an object, or in a very narrow set of cases such as building a domain specific language.

What this means concretely is that if you rename `ParseSqsMessage` to `SqsMessageParser`, and `apply` to `parse`, you preserve that general notation of having `What.action` or in your case `SqsMessageParser.parse`. This makes it dead obvious when reading where the function is called, whereas in its current form it could be either a function call or an object construction.  This only applies to static methods (any method on a companion object for instance).

Reply:
That makes sense. I was a little bit unsure about the best way to do this so more than happy to change